docs/sql-programming-guide.md

```diff
@@ -12,130 +12,130 @@ title: Spark SQL and DataFrames
-of its decedents. To create a basic `SQLContext`, all you need is a SparkContext.
+The entry point into all relational functionality in Spark is the [`SparkSession`](api/python/pyspark.sql.html#pyspark.sql.SparkSession) class. To create a basic `SparkSession`, just use `SparkSession.build`:
 
 {% highlight python %}
-from pyspark.sql import SQLContext
-sqlContext = SQLContext(sc)
+from pyspark.sql import SparkSession
+spark = SparkSession.build \
+  .master("local") \
+  .appName("Word Count") \
+  .config("spark.some.config.option", "some-value") \
+  .getOrCreate()
 {% endhighlight %}
 
 </div>
 
 <div data-lang="r"  markdown="1">
 
 The entry point into all relational functionality in Spark is the
-`SQLContext` class, or one of its decedents. To create a basic `SQLContext`, all you need is a SparkContext.
+`SparkSession` class. To create a basic `SparkSession`, all you need is a `SparkContext`.
 
 {% highlight r %}
-sqlContext <- sparkRSQL.init(sc)
+spark <- sparkRSQL.init(sc)
 {% endhighlight %}
 
 </div>
 </div>
 
-In addition to the basic `SQLContext`, you can also create a `HiveContext`, which provides a
-superset of the functionality provided by the basic `SQLContext`. Additional features include
-the ability to write queries using the more complete HiveQL parser, access to Hive UDFs, and the
-ability to read data from Hive tables. To use a `HiveContext`, you do not need to have an
-existing Hive setup, and all of the data sources available to a `SQLContext` are still available.
-`HiveContext` is only packaged separately to avoid including all of Hive's dependencies in the default
-Spark build. If these dependencies are not a problem for your application then using `HiveContext`
-is recommended for the 1.3 release of Spark. Future releases will focus on bringing `SQLContext` up
-to feature parity with a `HiveContext`.
+`SparkSession` in Spark 2.0 provides builtin support for Hive features including the ability to
+write queries using HiveQL, access to Hive UDFs, and the ability to read data from Hive tables.
+To use these features, you do not need to have an existing Hive setup, and all of the data sources
+available to a `SparkSession` are still available. These Hive features are only packaged separately
+to avoid including all of Hive's dependencies in the default Spark build.
 
 
 ## Creating DataFrames
 
-With a `SQLContext`, applications can create `DataFrame`s from an <a href='#interoperating-with-rdds'>existing `RDD`</a>, from a Hive table, or from <a href='#data-sources'>data sources</a>.
+With a `SparkSession`, applications can create DataFrames from an <a href='#interoperating-with-rdds'>existing `RDD`</a>,
+from a Hive table, or from <a href='#data-sources'>data sources</a>.
 
-As an example, the following creates a `DataFrame` based on the content of a JSON file:
+As an example, the following creates a DataFrame based on the content of a JSON file:
```

Comment:
based on the content of a JSON file => based on JSON data source API? 
So that this example is an ellaboration of 

```
With a `SparkSession`, applications can create DataFrames from an existing `RDD`,  from a Hive table, or from data sources
```